x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -136,8 +136,8 @@ export class TelemetryEventsSender {
     if (!this.core) {
       throw Error('could not fetch diagnostic alerts. core is not available');
     }
-    const callCluster = this.core.elasticsearch.legacy.client.callAsInternalUser;
-    return callCluster('search', query);
+    const esClient = this.core.elasticsearch.client;
```

Comment:
FWIW, there is an `esClient` I introduced in the latest Endpoint telemetry that is on the sender object itself.

```
this.esClient = core?.elasticsearch.client.asInternalUser;
```
https://github.com/elastic/kibana/pull/102171/files#diff-42eb463cfaca4b263f4d6543fd550415c343c5f004a3c72312c8e0dd704ef1caR86

What do you think is the more pragmatic way of doing this going forward? I'm happy for you to set the direction here.

Reply:
I think I will change it to what you have above to keep this part consistent. I think either pattern is fine and will keep the existing one you have.